lib/rsk-utils.js

```diff
@@ -529,6 +529,30 @@ const setFeePerKb = async (rskTxHelper, feePerKbInSatoshis) => {
 
 };
 
+/**
+ * Waits for the rsk mempool to get `atLeastExpectedCount` txs. Attempting `maxAttempts` times, checking every `waitTimeInMilliseconds`.
+ * @param {RskTransactionHelper} rskTxHelper 
+ * @param {number} atLeastExpectedCount 
+ * @param {number} waitTimeInMilliseconds defaults to 500 milliseconds
+ * @param {number} maxAttempts defaults to 20 attempts
+ * @returns {Promise<void>}
+ * @throws {Error} if the rsk mempool doesn't get at least `atLeastExpectedCount` txs after `maxAttempts` attempts
+ */
+const waitForRskMempoolToGetAtLeastThisManyTxs = async (rskTxHelper, atLeastExpectedCount, waitTimeInMilliseconds = 500, maxAttempts = 20) => {
+  let attempts = 1;
+  while(attempts <= maxAttempts) {
```

Comment:
It would then look like the following if we wanted to have access to the `attempts` variable:

```js
const waitForRskMempoolToGetAtLeastThisManyTxs = async (rskTxHelper, atLeastExpectedCount, waitTimeInMilliseconds = 500, maxAttempts = 20) => {
  let attempts = 1;
  for(; attempts <= maxAttempts; attempts++) {
    const rskMempoolTxs = await getRskMempoolTransactionHashes(rskTxHelper);
    logger.debug(`[${waitForRskMempoolToGetAtLeastThisManyTxs.name}] rsk mempool txs: ${rskMempoolTxs.length}`);
    if(rskMempoolTxs.length >= atLeastExpectedCount) {
      logger.debug(`[${waitForRskMempoolToGetAtLeastThisManyTxs.name}] rsk mempool has ${rskMempoolTxs.length} (at least expected was ${atLeastExpectedCount}) txs after ${attempts} attempts`);
      return;
    }
    await wait(waitTimeInMilliseconds);
  }

  throw new Error(`[${waitForRskMempoolToGetAtLeastThisManyTxs.name}] rsk mempool didn't get at least ${atLeastExpectedCount} txs after ${maxAttempts} attempts`);
};
```

By declaring it in the `for` like `for(let attemptt...`, we wouldn't have access it to it below the `for`. And with it declared above the `for`, then there's not really much difference than with the `while`, in fact, looks weirder this part like that.

Reply:
I would vote to keep as it is now.

Reply:
I see, looks good to me